include/dxc/Support/dxcapi.use.h

```diff
@@ -19,9 +19,52 @@ namespace dxc {
 extern const char *kDxCompilerLib;
 extern const char *kDxilLib;
 
+// Interface for common dll operations
+class DllLoader {
+public:
+  DllLoader() = default;
+
+  DllLoader(DllLoader &&other) = default;
+
+  virtual ~DllLoader() = default;
+
+  virtual HRESULT Initialize() = 0;
+
+  virtual HRESULT InitializeForDll(LPCSTR dll, LPCSTR entryPoint) = 0;
```

Comment:
It would be nice if we could remove `Initialize()` from the interface here and force the creator of the object to initialize it themselves. There are a couple of places that might be a bit awkward to update to do that, but I think it'd be cleaner if users of this class couldn't just arbitrarily reinitialize it in a way that could very much be confusing.

`InitializeForDll` would be harder to get rid of, but if we were to rename it to `overrideDll` or something like that I think it would make it clearer that it's there as an explicit hook to override which Dll you're using.